sys/linux/socket_tipc_netlink.txt

```diff
@@ -93,6 +93,12 @@ tipc_name_table_query {
 	upbound		int32be
 }
 
+tipc_aead_key {
+	alg_name	array[int8, 32]
+	keylen		bytesize[key, int16be]
```

Comment:
Updated. I used int16be because the len of the tipc_tlv_desc (in the same fiile) is set to int16be. Not sure if there's a connection there. It also seems that I can't set the range of a "len" type object, as it gives the error "base type argument has subargs"

Reply:
Right, but you _can_ set the range on `key` since it's an array.

Reply:
'len' is always computed as length of the target field. Range can't be used with 'len' because it won't be used anyway. If you want a random integer in a particular range, then use `int16be` instead of `len`, e.g. `int16be[5:10]`.

Reply:
This field is an `unsigned int` that is the length of the `key` field.   I think the correct description is `keylen     len[key, int32]`.

Reply:
I think len and bytesize would give us the same result here since it's a byte array?

Reply:
Yes, either `len` or `bytesize` will work.